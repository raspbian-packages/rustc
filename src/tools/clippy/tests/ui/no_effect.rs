#![feature(plugin, box_syntax, inclusive_range_syntax)]
#![plugin(clippy)]

#![warn(no_effect, unnecessary_operation)]
#![allow(dead_code)]
#![allow(path_statements)]
#![allow(deref_addrof)]
#![feature(untagged_unions)]

struct Unit;
struct Tuple(i32);
struct Struct {
    field: i32
}
enum Enum {
    Tuple(i32),
    Struct { field: i32 },
}

union Union {
    a: u8,
    b: f64,
}

fn get_number() -> i32 { 0 }
fn get_struct() -> Struct { Struct { field: 0 } }

unsafe fn unsafe_fn() -> i32 { 0 }

fn main() {
    let s = get_struct();
    let s2 = get_struct();

    0;
    s2;
    Unit;
    Tuple(0);
    Struct { field: 0 };
    Struct { ..s };
    Union { a: 0 };
    Enum::Tuple(0);
    Enum::Struct { field: 0 };
    5 + 6;
    *&42;
    &6;
    (5, 6, 7);
    box 42;
    ..;
    5..;
    ..5;
    5..6;
    5...6;
    [42, 55];
    [42, 55][1];
    (42, 55).1;
    [42; 55];
    [42; 55][13];
    let mut x = 0;
    || x += 5;

    // Do not warn
    get_number();
    unsafe { unsafe_fn() };

    Tuple(get_number());
    Struct { field: get_number() };
    Struct { ..get_struct() };
    Enum::Tuple(get_number());
    Enum::Struct { field: get_number() };
    5 + get_number();
    *&get_number();
    &get_number();
    (5, 6, get_number());
    box get_number();
    get_number()..;
    ..get_number();
    5..get_number();
    [42, get_number()];
    [42, 55][get_number() as usize];
    (42, get_number()).1;
    [get_number(); 55];
    [42; 55][get_number() as usize];
    {get_number()};
}
