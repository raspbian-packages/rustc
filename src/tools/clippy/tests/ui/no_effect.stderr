error: statement with no effect
  --> $DIR/no_effect.rs:34:5
   |
34 |     0;
   |     ^^
   |
   = note: `-D no-effect` implied by `-D warnings`

error: statement with no effect
  --> $DIR/no_effect.rs:35:5
   |
35 |     s2;
   |     ^^^

error: statement with no effect
  --> $DIR/no_effect.rs:36:5
   |
36 |     Unit;
   |     ^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:37:5
   |
37 |     Tuple(0);
   |     ^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:38:5
   |
38 |     Struct { field: 0 };
   |     ^^^^^^^^^^^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:39:5
   |
39 |     Struct { ..s };
   |     ^^^^^^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:40:5
   |
40 |     Union { a: 0 };
   |     ^^^^^^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:41:5
   |
41 |     Enum::Tuple(0);
   |     ^^^^^^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:42:5
   |
42 |     Enum::Struct { field: 0 };
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:43:5
   |
43 |     5 + 6;
   |     ^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:44:5
   |
44 |     *&42;
   |     ^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:45:5
   |
45 |     &6;
   |     ^^^

error: statement with no effect
  --> $DIR/no_effect.rs:46:5
   |
46 |     (5, 6, 7);
   |     ^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:47:5
   |
47 |     box 42;
   |     ^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:48:5
   |
48 |     ..;
   |     ^^^

error: statement with no effect
  --> $DIR/no_effect.rs:49:5
   |
49 |     5..;
   |     ^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:50:5
   |
50 |     ..5;
   |     ^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:51:5
   |
51 |     5..6;
   |     ^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:52:5
   |
52 |     5...6;
   |     ^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:53:5
   |
53 |     [42, 55];
   |     ^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:54:5
   |
54 |     [42, 55][1];
   |     ^^^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:55:5
   |
55 |     (42, 55).1;
   |     ^^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:56:5
   |
56 |     [42; 55];
   |     ^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:57:5
   |
57 |     [42; 55][13];
   |     ^^^^^^^^^^^^^

error: statement with no effect
  --> $DIR/no_effect.rs:59:5
   |
59 |     || x += 5;
   |     ^^^^^^^^^^

error: statement can be reduced
  --> $DIR/no_effect.rs:65:5
   |
65 |     Tuple(get_number());
   |     ^^^^^^^^^^^^^^^^^^^^ help: replace it with: `get_number();`
   |
   = note: `-D unnecessary-operation` implied by `-D warnings`

error: statement can be reduced
  --> $DIR/no_effect.rs:66:5
   |
66 |     Struct { field: get_number() };
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace it with: `get_number();`

error: statement can be reduced
  --> $DIR/no_effect.rs:67:5
   |
67 |     Struct { ..get_struct() };
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace it with: `get_struct();`

error: statement can be reduced
  --> $DIR/no_effect.rs:68:5
   |
68 |     Enum::Tuple(get_number());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace it with: `get_number();`

error: statement can be reduced
  --> $DIR/no_effect.rs:69:5
   |
69 |     Enum::Struct { field: get_number() };
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace it with: `get_number();`

error: statement can be reduced
  --> $DIR/no_effect.rs:70:5
   |
70 |     5 + get_number();
   |     ^^^^^^^^^^^^^^^^^ help: replace it with: `5;get_number();`

error: statement can be reduced
  --> $DIR/no_effect.rs:71:5
   |
71 |     *&get_number();
   |     ^^^^^^^^^^^^^^^ help: replace it with: `get_number();`

error: statement can be reduced
  --> $DIR/no_effect.rs:72:5
   |
72 |     &get_number();
   |     ^^^^^^^^^^^^^^ help: replace it with: `get_number();`

error: statement can be reduced
  --> $DIR/no_effect.rs:73:5
   |
73 |     (5, 6, get_number());
   |     ^^^^^^^^^^^^^^^^^^^^^ help: replace it with: `5;6;get_number();`

error: statement can be reduced
  --> $DIR/no_effect.rs:74:5
   |
74 |     box get_number();
   |     ^^^^^^^^^^^^^^^^^ help: replace it with: `get_number();`

error: statement can be reduced
  --> $DIR/no_effect.rs:75:5
   |
75 |     get_number()..;
   |     ^^^^^^^^^^^^^^^ help: replace it with: `get_number();`

error: statement can be reduced
  --> $DIR/no_effect.rs:76:5
   |
76 |     ..get_number();
   |     ^^^^^^^^^^^^^^^ help: replace it with: `get_number();`

error: statement can be reduced
  --> $DIR/no_effect.rs:77:5
   |
77 |     5..get_number();
   |     ^^^^^^^^^^^^^^^^ help: replace it with: `5;get_number();`

error: statement can be reduced
  --> $DIR/no_effect.rs:78:5
   |
78 |     [42, get_number()];
   |     ^^^^^^^^^^^^^^^^^^^ help: replace it with: `42;get_number();`

error: statement can be reduced
  --> $DIR/no_effect.rs:79:5
   |
79 |     [42, 55][get_number() as usize];
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace it with: `[42, 55];get_number() as usize;`

error: statement can be reduced
  --> $DIR/no_effect.rs:80:5
   |
80 |     (42, get_number()).1;
   |     ^^^^^^^^^^^^^^^^^^^^^ help: replace it with: `42;get_number();`

error: statement can be reduced
  --> $DIR/no_effect.rs:81:5
   |
81 |     [get_number(); 55];
   |     ^^^^^^^^^^^^^^^^^^^ help: replace it with: `get_number();`

error: statement can be reduced
  --> $DIR/no_effect.rs:82:5
   |
82 |     [42; 55][get_number() as usize];
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: replace it with: `[42; 55];get_number() as usize;`

error: statement can be reduced
  --> $DIR/no_effect.rs:83:5
   |
83 |     {get_number()};
   |     ^^^^^^^^^^^^^^^ help: replace it with: `get_number();`

error: aborting due to 44 previous errors

